scripts/battlefields/Bearclaw_Pinnacle/when_hell_freezes_over.lua

```diff
@@ -0,0 +1,119 @@
+    battlefieldId    = xi.battlefield.id.WHEN_HELL_FREEZES_OVER,
+    maxPlayers       = 18,
+    levelCap         = 75,
+    timeLimit        = utils.minutes(15),
+    index            = 2,
+    entryNpc         = 'Wind_Pillar_3',
+    exitNpc          = 'Wind_Pillar_Exit',
+    requiredKeyItems = { xi.ki.ZEPHYR_FAN, message = bearclawID.text.ZEPHYR_RIPS },
+    grantXP          = 3000,
+})
+
+-- Spawning function for this battlefield. Spawns a wave of 1-3 Snow Devils of either all BLM or all WAR
+
+local function spawnSnowDevils(battlefield)
+    local base   = bearclawID.mob.SNOW_DEVIL + (battlefield:getArea() - 1) * 8
+    local offset = math.random(0, 1) * 3
+    local count  = math.random(0, 2)
+
+    for i = 0, count do
+        local id  = base + offset + i
+        local mob = GetMobByID(id)
+
+        if mob then
+            mob:spawn()
+            local players = battlefield:getPlayers()
+            local wave  = battlefield:getLocalVar('wave')
+            if wave >= 1 then
+                for _, player in pairs(players) do
+                    if player:isAlive() then
+                        mob:updateEnmity(player)
+                        break
+                    end
+                end
+            end
+        end
+    end
+end
+
+-- Defines the monsters contained within this battlefield and defines a custom win condition. (4 waves completed)
+-- We set all mobs spawned to false, and handle the wave spawning logic in onBattlefieldTick.
+-- Any logic for spawning monsters in waves should be handled outside of content.groups to avoid them passing through initialization.
+
+content.groups = {
+    {
+        mobs = { 'Snow_Devil_war', 'Snow_Devil_blm' },
+        spawned = false,
+
+        allDeath = function(battlefield, mob)
+            local wave = battlefield:getLocalVar('wave') + 1
+            if wave > 4 then
+                content:handleAllMonstersDefeated(battlefield, mob)
+            else
+                battlefield:setLocalVar('wave', wave)
+                battlefield:setLocalVar('nextWaveDelay', GetSystemTime() + 5)
+            end
+        end,
+    },
+}
+
+-- Handles wave spawning and delays
+-- We call back Battlefield.onBattlefieldTick(self, battlefield, tick) to preserve the base functionality. (Many things will break without doing this)
+
+function content:onBattlefieldTick(battlefield, tick)
+    Battlefield.onBattlefieldTick(self, battlefield, tick)
+
+    local now   = GetSystemTime()
+    local wave  = battlefield:getLocalVar('wave')
+    local delay = battlefield:getLocalVar('nextWaveDelay')
+
+    -- Initial wave spawn
+    if wave == 0 then
+        spawnSnowDevils(battlefield)
+        battlefield:setLocalVar('wave', 1)
+        battlefield:setLocalVar('nextWaveDelay', 0)
+        return
+    end
+
+    -- Spawns next wave after a 20 second delay
+    if delay > 0 and now >= delay then
+        battlefield:setLocalVar('nextWaveDelay', 0)
+        spawnSnowDevils(battlefield)
+    end
```

Comment:
This smells. If waves are time-based, the function to spawn them would need a check to see if the mob is not spawned already